website_version_ce/models/experiment.py

```diff
@@ -0,0 +1,218 @@
+                        ('experiment_id.website_id', '=',
+                         exp_ver.experiment_id.website_id.id),
+                        ('experiment_id.state', '=', 'running')
+                    ]):
+                        raise ValidationError(_(
+                            'This experiment contains a view which '
+                            'is already used in another running experience'))
+        return True
+
+    @api.multi
+    @api.constrains('website_id', 'experiment_version_ids')
+    def _check_website(self):
+        for exp in self:
+            for exp_ver in exp.experiment_version_ids:
+                if not exp_ver.version_id.website_id.id == exp.website_id.id:
+                    raise ValidationError(_(
+                        'This experiment must have versions which '
+                        'are in the same website'))
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None,
+                   context=None, orderby=False, lazy=True):
+        """ Override read_group to always display all states. """
+        if groupby and groupby[0] == "state":
+            # Default result structure
+            states = [('running', 'Running'), ('paused', 'Paused'),
+                      ('ended', 'Ended')]
+            read_group_all_states = [{
+                '__context': {'group_by': groupby[1:]},
+                '__domain': domain + [('state', '=', state_value)],
+                'state': state_value,
+                'state_count': 0,
+            } for state_value, state_name in states]
+            # Get standard results
+            read_group_res = super(Experiment, self).read_group(
+                domain, fields, groupby, offset=offset, limit=limit,
+                context=context, orderby=orderby, lazy=lazy)
+            # Update standard results with default results
+            result = []
+            for state_value, state_name in states:
+                res = filter(lambda x: x['state'] == state_value,
+                             read_group_res)
+                if not res:
+                    res = filter(lambda x: x['state'] == state_value,
+                                 read_group_all_states)
+                res[0]['state'] = [state_value, state_name]
+                result.append(res[0])
+            return result
+        else:
+            return super(Experiment, self).read_group(
+                domain, fields, groupby, offset=offset, limit=limit,
+                context=context, orderby=orderby, lazy=lazy)
+
+    @api.multi
+    def _get_version_number(self):
+        for exp in self:
+            exp.version_number = len(exp.experiment_version_ids) + 1
+
+    @api.model
+    def create(self, vals):
+        exp = {
+            'name': vals['name'],
+            'objectiveMetric': self.env['website_version_ce.goals'].browse(
+                [vals['goal_id']]).google_ref,
```

Comment:
Remove external square brackets in the `.browse(...)` params. It should be `.browse(vals['goal_id'])`